tests/predict/test_predict.py

```diff
@@ -209,14 +212,81 @@ class OutputOnlySignature(dspy.Signature):
     assert lm.get_convo(-1) == textwrap.dedent(
         """\
         Given the fields , produce the fields `output`.
-        
+
         ---
-        
+
         Follow the following format.
-        
+
         Output: ${output}
-        
+
         ---
-        
+
         Output: short answer"""
     )
+
+
+@pytest.fixture(name="SandwichIdea")
+def sandwich_idea_signature():
+    class SandwichIdea(dspy.Signature):
+        """Based on the meal and dietary requirements, suggest a sandwich idea."""
+
+        meal: str = dspy.InputField()
+        dietary_requiements: str = dspy.InputField()
+        bread: str = dspy.OutputField()
+        protein: str = dspy.OutputField()
+        fat: str = dspy.OutputField()
+        garnish: str = dspy.OutputField()
+        sauce: str = dspy.OutputField()
+
+    return SandwichIdea
+
+
+def test_extend_generation(SandwichIdea):
+    lm = DummyLM(
+        [
+            " whole wheat\n\nProtein: turkey\n\nFat: avocado",
+            " tomato\n\nSauce: mustard",
+        ]
+    )
+    dspy.settings.configure(lm=lm)
+
+    prediction = Predict(SandwichIdea)(meal="lunch", dietary_requiements="N/A")
+    assert prediction.bread == "whole wheat"
+    assert prediction.protein == "turkey"
+    assert prediction.fat == "avocado"
+    assert prediction.garnish == "tomato"
+    assert prediction.sauce == "mustard"
```

Comment:
@okhat - I'm not sure about the other tests I've added, but I was definitely expecting this test to pass. [However](https://github.com/stanfordnlp/dspy/actions/runs/9577937593/job/26407127842?pr=1172#step:8:48):

```python
SandwichIdea = SandwichIdea(meal, dietary_requiements -> bread, protein, fat, garnish, sauce
    instructions='Based on the meal and ...notation=str required=True json_schema_extra={'__dspy_field_type': 'output', 'prefix': 'Sauce:', 'desc': '${sauce}'})
)

    def test_extend_generation(SandwichIdea):
        lm = DummyLM(
            [
                " whole wheat\n\nProtein: turkey\n\nFat: avocado",
                " tomato\n\nSauce: mustard",
            ]
        )
        dspy.settings.configure(lm=lm)
    
        prediction = Predict(SandwichIdea)(meal="lunch", dietary_requiements="N/A")
        assert prediction.bread == "whole wheat"
        assert prediction.protein == "turkey"
        assert prediction.fat == "avocado"
>       assert prediction.garnish == "tomato"
E       AssertionError: assert '' == 'tomato'
E         - tomato
```

I think when "" is added as the result of the field on https://github.com/stanfordnlp/dspy/blob/34725d08e337fef293b5a2126cf7bf8abb80e41a/dsp/primitives/predict.py#L98 it is interperated as the final value by the extract method of template and the actual value is not included correctly.